### DISCUSSION TITLE
Add castBoolean

### DISCUSSION THREAD

**Discussion by MarlonPassos-git** [radashi-org/radashi#236] [Ideas]
Recently, I encountered an issue while parsing environment variables in a project where the values could come in different forms, such as "True", "true", "", "FALSE", "false", etc. The task was simple: convert these values into booleans consistently. At that moment, I recalled the [ParseBool](https://pkg.go.dev/strconv#ParseBool) function from Go, which handles this kind of conversion perfectly.

I believe a similar function could be a valuable addition to Radashi. It would provide a standardized and reliable way to handle such conversions.

Interestingly, I found a [Zod issue](https://github.com/colinhacks/zod/issues/2985) requesting a similar feature, which highlights the necessity and utility of such a feature in libraries dealing with data validation and parsing.

it would be an adaptation for Boolean()

```
castBoolen("true");    // true
castBoolen("false");   // false
castBoolen("True");    // true
castBoolen("False");   // false
castBoolen("tRue");    // true
castBoolen("fAlse");   // false
castBoolen("");        // false 
castBoolen(undefined); // false 
b1.parse("others");  // false
```

I'm open to other name suggestions

**Comment by aleclarson** [comment]
If it only accepts a `string` type, then I'd say go with `parseBoolean`.

If we wanted it to be a “better `Boolean()`”, then `castBoolean` works for me. In this case, strings are parsed, other types are true if truthy else false. Though I'm weary of such an implementation, as no use cases have popped into my mind that would require it. Therefore, I'm leaning toward the simpler `parseBoolean`.

As far as implementation goes, here's my proposal:
```ts
function parseBoolean(str: string): boolean | undefined {
  const match = /^(?:(1|true)||0|false)$/i.exec(str)
  return match && match[1] !== undefined
}
```
It returns…
- true for `"1"` and `"true"` (case insensitive)
- false for `""`, `"0"` and `"false"` (again, case insensitive)
- else null

You can use the `??` operator to set a default value:

```ts
parseBoolean(str) ?? true
```

You can use type assertion for non-string inputs (though it's not recommended):

```ts
parseBoolean(true as any)      // => true
parseBoolean(false as any)     // => false
parseBoolean(1 as any)         // => true
parseBoolean(0 as any)         // => false
parseBoolean(-0 as any)        // => false, because String(-0) === "0"
parseBoolean(-1 as any)        // => null
parseBoolean(null as any)      // => null
parseBoolean(undefined as any) // => null
```

**Reply by MarlonPassos-git** [reply]
It would be a case of dealing only with strings. I like `parseBoolean`.
